test/Microsoft.AspNet.Mvc.Core.Test/Internal/UrlUtilityTest.cs

```diff
@@ -0,0 +1,168 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.Internal
+{
+    public class UrlUtilityTest
+    {
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        public void IsLocalUrl_ReturnsFalseOnEmpty(string url)
+        {
+            // Arrange && Act
+            var result = UrlUtility.IsLocalUrl(url);
+
+            // Assert
+            Assert.False(result);
+        }
+
+        [Theory]
+        [InlineData("/foo.html")]
+        [InlineData("/www.example.com")]
+        [InlineData("/")]
+        public void IsLocalUrl_AcceptsRootedUrls(string url)
+        {
+            // Arrange && Act
+            var result = UrlUtility.IsLocalUrl(url);
+
+            // Assert
+            Assert.True(result);
+        }
+
+        [Theory]
+        [InlineData("~/")]
+        [InlineData("~/foo.html")]
+        public void IsLocalUrl_AcceptsApplicationRelativeUrls(string url)
+        {
+            // Arrange && Act
+            var result = UrlUtility.IsLocalUrl(url);
+
+            // Assert
+            Assert.True(result);
+        }
+
+        [Theory]
+        [InlineData("foo.html")]
+        [InlineData("../foo.html")]
+        [InlineData("fold/foo.html")]
+        public void IsLocalUrl_RejectsRelativeUrls(string url)
+        {
+            // Arrange && Act
+            var result = UrlUtility.IsLocalUrl(url);
+
+            // Assert
+            Assert.False(result);
+        }
+
+        [Theory]
+        [InlineData("http:/foo.html")]
+        [InlineData("hTtP:foo.html")]
+        [InlineData("http:/www.example.com")]
+        [InlineData("HtTpS:/www.example.com")]
+        public void IsLocalUrl_RejectValidButUnsafeRelativeUrls(string url)
+        {
+            // Arrange && Act
+            var result = UrlUtility.IsLocalUrl(url);
+
+            // Assert
+            Assert.False(result);
+        }
+
+        [Theory]
+        [InlineData("http://www.mysite.com/appDir/foo.html")]
+        [InlineData("http://WWW.MYSITE.COM")]
+        public void IsLocalUrl_RejectsUrlsOnTheSameHost(string url)
```

Comment:
@yishaigalatzer - the naming of this is historical cruft. These tests and code were copied almost verbatim for reasons.